drivers/sensor/meas/ms5837/ms5837.c

```diff
@@ -348,34 +348,26 @@ static int ms5837_init(const struct device *dev)
 		return err;
 	}
 
-	const int type_id = (data->factory >> 5) & 0x7f;
-
-	switch (type_id) {
-	case  MS5837_02BA01:
-	case MS5837_02BA21:
-		data->comp_func = ms5837_compensate_02;
-		break;
-	case MS5837_30BA26:
-		data->comp_func = ms5837_compensate_30;
-		break;
-	default:
-		LOG_WRN(" unrecognized type: '%2x', defaulting to MS5837-30", type_id);
-		data->comp_func = ms5837_compensate_30;
-		break;
-	}
-
 	return 0;
 }
 
-#define MS5837_DEFINE(inst)								\
-	static struct ms5837_data ms5837_data_##inst;					\
-											\
-	static const struct ms5837_config ms5837_config_##inst = {			\
-		.i2c = I2C_DT_SPEC_INST_GET(inst),					\
-	};										\
-											\
-	SENSOR_DEVICE_DT_INST_DEFINE(inst, ms5837_init, NULL,				\
-			      &ms5837_data_##inst, &ms5837_config_##inst, POST_KERNEL,	\
-			      CONFIG_SENSOR_INIT_PRIORITY, &ms5837_api_funcs);		\
-
-DT_INST_FOREACH_STATUS_OKAY(MS5837_DEFINE)
+#define MS5837_DEFINE(inst, type)                                                                  \
+	static struct ms5837_data ms5837_data_##inst = {                                           \
+		.comp_func = ms5837_compensate_##type                                              \
```

Comment:
If this is done statically (which I like a lot and is much better) it should go into the config not the data.

Reply:
Good point. I fixed it and pushed.